src/structural_elements/CCPACSFuselageWallSegment.cpp

```diff
@@ -0,0 +1,292 @@
+/* 
+* Copyright (C) 2019 German Aerospace Center (DLR/SC)
+*
+* Created: 2019-05-14 Martin Siggel <Martin.Siggel@dlr.de> with Marko Alder <Marko.Alder@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "CCPACSFuselageWallSegment.h"
+#include "generated/CPACSWallSegments.h"
+#include "generated/CPACSWalls.h"
+#include "CCPACSFuselageStructure.h"
+#include "CCPACSFuselage.h"
+#include "CCPACSFuselageSegment.h"
+#include "CNamedShape.h"
+#include "CCPACSWallPosition.h"
+#include "CTiglUIDManager.h"
+#include "tiglcommonfunctions.h"
+#include "to_string.h"
+
+#include <Bnd_Box.hxx>
+#include <BRepBndLib.hxx>
+#include <BRep_Tool.hxx>
+#include <BRepBuilderAPI_MakeEdge.hxx>
+#include <BRepBuilderAPI_MakeWire.hxx>
+#include <BRepBuilderAPI_MakeFace.hxx>
+#include <TopExp_Explorer.hxx>
+#include <TopoDS.hxx>
+#include <GeomLib_IsPlanarSurface.hxx>
+#include <IntCurvesFace_ShapeIntersector.hxx>
+#include <TopTools_IndexedMapOfShape.hxx>
+#include <TopExp.hxx>
+#include <gp_Ax3.hxx>
+#include <gp_Pln.hxx>
+#include <BRepClass3d_SolidClassifier.hxx>
+
+#include <algorithm>
+
+namespace {
+
+// create a face out of four points
+TopoDS_Face MakeFace(gp_Pnt const& p1, gp_Pnt const& p2, gp_Pnt const& p3, gp_Pnt const& p4) {
+    TopoDS_Edge e1 = BRepBuilderAPI_MakeEdge(p1, p2).Edge();
+    TopoDS_Edge e2 = BRepBuilderAPI_MakeEdge(p2, p3).Edge();
+    TopoDS_Edge e3 = BRepBuilderAPI_MakeEdge(p3, p4).Edge();
+    TopoDS_Edge e4 = BRepBuilderAPI_MakeEdge(p4, p1).Edge();
+    TopoDS_Wire w = BRepBuilderAPI_MakeWire(e1, e2, e3, e4).Wire();
+    return BRepBuilderAPI_MakeFace(w, false).Face();
+}
```

Comment:
Please remove and use

https://github.com/DLR-SC/tigl/blob/685ee62778ab727bb05a93ddac51a092f29ecd80/src/common/tiglcommonfunctions.h#L159

instead.